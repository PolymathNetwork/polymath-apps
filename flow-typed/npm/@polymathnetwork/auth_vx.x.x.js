// flow-typed signature: d731cb9b18720f4cf204447fa5d4d547
// flow-typed version: <<STUB>>/@polymathnetwork/auth_v0.0.0/flow_v0.81.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@polymathnetwork/auth'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@polymathnetwork/auth' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@polymathnetwork/auth/babel.config' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/actions' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/helpers' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/index' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/networks' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/PolymathAuth' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/src/reducer' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/webpack.dev' {
  declare module.exports: any;
}

declare module '@polymathnetwork/auth/webpack.prod' {
  declare module.exports: any;
}

// Filename aliases
declare module '@polymathnetwork/auth/babel.config.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/babel.config'>;
}
declare module '@polymathnetwork/auth/src/actions.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/actions'>;
}
declare module '@polymathnetwork/auth/src/helpers.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/helpers'>;
}
declare module '@polymathnetwork/auth/src/index.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/index'>;
}
declare module '@polymathnetwork/auth/src/networks.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/networks'>;
}
declare module '@polymathnetwork/auth/src/PolymathAuth.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/PolymathAuth'>;
}
declare module '@polymathnetwork/auth/src/reducer.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/src/reducer'>;
}
declare module '@polymathnetwork/auth/webpack.dev.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/webpack.dev'>;
}
declare module '@polymathnetwork/auth/webpack.prod.js' {
  declare module.exports: $Exports<'@polymathnetwork/auth/webpack.prod'>;
}
