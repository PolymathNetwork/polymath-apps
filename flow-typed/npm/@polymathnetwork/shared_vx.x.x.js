// flow-typed signature: 4bd024aac82bd9d25d925fc8faf72b97
// flow-typed version: <<STUB>>/@polymathnetwork/shared_v0.0.0/flow_v0.81.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@polymathnetwork/shared'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@polymathnetwork/shared' {
  declare module.exports: any;
}
