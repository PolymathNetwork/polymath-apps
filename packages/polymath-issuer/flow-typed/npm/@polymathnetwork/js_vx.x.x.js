// flow-typed signature: f74591e4e08edd90ce9c87e65e769c1d
// flow-typed version: <<STUB>>/@polymathnetwork/js_v0.0.0/flow_v0.81.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@polymathnetwork/js'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@polymathnetwork/js' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@polymathnetwork/js/babel.config' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CappedSTOFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/Contract' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CountTransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CountTransferManagerFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/IModuleFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PercentageTransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PermissionManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PolyToken' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/SecurityToken' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/SecurityTokenRegistry' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/STO' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/TickerRegistry' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/TransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/index' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/IPFS' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/types' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/webpack.dev' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/webpack.prod' {
  declare module.exports: any;
}

// Filename aliases
declare module '@polymathnetwork/js/babel.config.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/babel.config'>;
}
declare module '@polymathnetwork/js/src/contracts/CappedSTOFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/CappedSTOFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/Contract.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/Contract'>;
}
declare module '@polymathnetwork/js/src/contracts/CountTransferManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/CountTransferManager'>;
}
declare module '@polymathnetwork/js/src/contracts/CountTransferManagerFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/CountTransferManagerFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/IModuleFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/IModuleFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/PercentageTransferManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PercentageTransferManager'>;
}
declare module '@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/PermissionManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PermissionManager'>;
}
declare module '@polymathnetwork/js/src/contracts/PolyToken.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PolyToken'>;
}
declare module '@polymathnetwork/js/src/contracts/SecurityToken.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/SecurityToken'>;
}
declare module '@polymathnetwork/js/src/contracts/SecurityTokenRegistry.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/SecurityTokenRegistry'>;
}
declare module '@polymathnetwork/js/src/contracts/STO.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/STO'>;
}
declare module '@polymathnetwork/js/src/contracts/TickerRegistry.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/TickerRegistry'>;
}
declare module '@polymathnetwork/js/src/contracts/TransferManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/TransferManager'>;
}
declare module '@polymathnetwork/js/src/index.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/index'>;
}
declare module '@polymathnetwork/js/src/IPFS.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/IPFS'>;
}
declare module '@polymathnetwork/js/src/types.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/types'>;
}
declare module '@polymathnetwork/js/webpack.dev.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/webpack.dev'>;
}
declare module '@polymathnetwork/js/webpack.prod.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/webpack.prod'>;
}
