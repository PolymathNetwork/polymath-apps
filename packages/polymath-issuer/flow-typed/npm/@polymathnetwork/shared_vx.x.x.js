// flow-typed signature: 1b60d8488df603980ab0f1307a4a306d
// flow-typed version: <<STUB>>/@polymathnetwork/shared_v0.0.0/flow_v0.81.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@polymathnetwork/shared'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@polymathnetwork/shared' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/core_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/@babel/node_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-proposal-class-properties_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-transform-runtime_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/@babel/preset-env_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/@babel/preset-flow_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/flow-bin_v0.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/ganache-cli_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/jest_v22.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/polymath-core_vx.x.x' {
  declare module.exports: any;
}

declare module '@polymathnetwork/shared/flow-typed/npm/truffle_vx.x.x' {
  declare module.exports: any;
}

// Filename aliases
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/core_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/core_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/node_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/node_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-proposal-class-properties_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-proposal-class-properties_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-transform-runtime_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/plugin-transform-runtime_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/preset-env_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/preset-env_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/@babel/preset-flow_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/@babel/preset-flow_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/flow-bin_v0.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/flow-bin_v0.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/ganache-cli_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/ganache-cli_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/jest_v22.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/jest_v22.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/polymath-core_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/polymath-core_vx.x.x'
  >;
}
declare module '@polymathnetwork/shared/flow-typed/npm/truffle_vx.x.x.js' {
  declare module.exports: $Exports<
    '@polymathnetwork/shared/flow-typed/npm/truffle_vx.x.x'
  >;
}
