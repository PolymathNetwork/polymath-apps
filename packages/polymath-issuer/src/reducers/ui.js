// @flow
import { LOCATION_CHANGE } from 'connected-react-router';

export type UIState = {
  header: {
    variant: 'default' | 'transparent',
  },
  footer: {
    variant: 'default' | 'transparent',
  },
};

const defaultState: UIState = {
  header: {
    variant: 'default',
  },
  footer: {
    variant: 'default',
  },
};

export default (state: UIState = defaultState, action) => {
  if (action.type === LOCATION_CHANGE) {
    const pathname = action.payload.location.pathname;

    if (pathname === '/') {
      return {
        ...state,
        header: {
          variant: 'transparent',
        },
        footer: {
          variant: 'transparent',
        },
      };
    }

    return defaultState;
  } else {
    return state;
  }
};
