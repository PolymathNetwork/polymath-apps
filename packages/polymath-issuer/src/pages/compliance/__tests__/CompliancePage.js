import { CompliancePage } from '../CompliancePage';
import DefaultCompliancePage from '../CompliancePage';
import renderer from 'react-test-renderer';
import React from 'react';
import configureStore from 'redux-mock-store';
import { theme, GlobalStyles } from '@polymathnetwork/ui';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import moment from 'moment-timezone';
import { Provider } from 'react-redux';
const mockStore = configureStore([]);
moment.tz.setDefault('America/New_York');

describe('Compliance Page', () => {
  it('should render not found when there is no token', () => {
    const props = {
      fetchWhitelist: jest.fn(),
      fetchManagers: jest.fn(),
      fetchPartialTransfers: jest.fn(),
      theme,
    };
    const tree = shallow(<CompliancePage {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const props = {
      fetchWhitelist: jest.fn(),
      fetchManagers: jest.fn(),
      fetchPartialTransfers: jest.fn(),
      theme,
      token: {
        address: '0x121212',
        contract: {
          version: '3.0.0',
        },
      },
    };
    const tree = shallow(<CompliancePage {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should not render RPTM tab when token version below 3.0.0', () => {
    const props = {
      fetchWhitelist: jest.fn(),
      fetchManagers: jest.fn(),
      fetchPartialTransfers: jest.fn(),
      theme,
      token: {
        address: '0x121212',
        contract: {
          version: '2.9.0',
        },
      },
    };
    const tree = shallow(<CompliancePage {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should call addPartialTM when toggle is set to true', async () => {
    const props = {
      fetchWhitelist: jest.fn(),
      fetchManagers: jest.fn(),
      fetchPartialTransfers: jest.fn(),
      addPartialTM: jest.fn(),
      archivePartialTM: jest.fn(),
      theme,
      token: {
        address: '0x121212',
        contract: {
          version: '3.0.0',
        },
      },
    };
    const tree = shallow(<CompliancePage {...props} />);
    await tree.instance().handleTogglePartialTransfer(true);
    expect(props.addPartialTM).toHaveBeenCalled();
    expect(props.archivePartialTM.mock.calls.length).toBe(0);
  });

  it('should call archivePartialTM when toggle is set to false', async () => {
    const props = {
      fetchWhitelist: jest.fn(),
      fetchManagers: jest.fn(),
      fetchPartialTransfers: jest.fn(),
      addPartialTM: jest.fn(),
      archivePartialTM: jest.fn(),
      theme,
      token: {
        address: '0x121212',
        contract: {
          version: '3.0.0',
        },
      },
    };
    const tree = shallow(<CompliancePage {...props} />);
    await tree.instance().handleTogglePartialTransfer(false);
    expect(props.archivePartialTM).toHaveBeenCalled();
    expect(props.addPartialTM.mock.calls.length).toBe(0);
  });

  it('should show PartialTransferTable when RPTM is enabled', () => {
    const props = {
      fetchWhitelist: jest.fn(),
      fetchManagers: jest.fn(),
      fetchPartialTransfers: jest.fn(),
      addPartialTM: jest.fn(),
      archivePartialTM: jest.fn(),
      theme,
      token: {
        address: '0x121212',
        contract: {
          version: '3.0.0',
        },
      },
    };
    const tree = shallow(<CompliancePage {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
