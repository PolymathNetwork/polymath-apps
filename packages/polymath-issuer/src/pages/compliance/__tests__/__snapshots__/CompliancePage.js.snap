// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compliance Page should render Approval Table when MATM is enabled 1`] = `
<Page
  py="xl"
  title="Compliance – Polymath"
>
  <Connect(Progress) />
  <Grid
    gridGap={4}
  >
    <GridRow
      gridGap="0px"
    >
      <GridCol
        gridSpan={9}
      >
        <h1
          className="pui-h1"
        >
          Token Compliance
        </h1>
        <h3
          className="pui-h3"
        >
          Manage compliance requirements of your tokens. You can import and export your whitelist, manage ownership percentages, add 3rd party whitelist managers, and manage approval of token transfers between wallets.
        </h3>
      </GridCol>
      <GridCol
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <Button
          className="import-whitelist-btn"
          disabled={false}
          icon="upload"
          iconDescription="Provide icon description if icon is used"
          kind="primary"
          onClick={[Function]}
          small={false}
          tabIndex={0}
          type="button"
        >
          Import Whitelist
        </Button>
      </GridCol>
      <GridCol
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <Connect(ImportWhitelistModal)
          isOpen={false}
          onClose={[Function]}
          onSubmit={[Function]}
        />
        <Button
          className="import-whitelist-btn"
          disabled={false}
          icon="download"
          iconDescription="Provide icon description if icon is used"
          kind="secondary"
          onClick={[Function]}
          small={false}
          tabIndex={0}
          type="button"
        >
          Export Whitelist
        </Button>
      </GridCol>
      <GridCol
        style={
          Object {
            "alignSelf": "center",
            "justifySelf": "left",
          }
        }
      >
        <OverflowMenu
          ariaLabel="list of options"
          direction="bottom"
          flipped={true}
          floatingMenu={true}
          iconDescription="open and close list of options"
          menuOffset={[Function]}
          menuOffsetFlip={[Function]}
          onClick={[Function]}
          onClose={[Function]}
          onKeyDown={[Function]}
          onOpen={[Function]}
          open={false}
          style={
            Object {
              "float": "right",
            }
          }
          tabIndex={0}
        >
          <OverflowMenuItem
            disabled={false}
            hasDivider={false}
            isDelete={false}
            itemText="Pause All Transfers"
            onClick={[Function]}
          />
        </OverflowMenu>
      </GridCol>
    </GridRow>
    <GridRow
      gridGap={4}
    >
      <GridCol
        gridSpan={
          Array [
            12,
            12,
            6,
            6,
          ]
        }
      >
        <div
          id="compliance"
        >
          <br />
          <div
            className="pui-page-box compliance-form"
          >
            <h1
              className="pui-h1"
            >
              Ownership Restrictions
            </h1>
            <div
              className="compliance-settings"
            >
              <div
                className="bx--form-item"
              >
                <label
                  className="bx--label"
                  htmlFor="percentageToggle"
                >
                  Enable Ownership Restrictions
                </label>
                <Toggle
                  defaultToggled={false}
                  id="percentageToggle"
                  labelA="Off"
                  labelB="On"
                  onToggle={[Function]}
                  toggled={false}
                />
              </div>
              <div
                className="bx--form-item"
                style={Object {}}
              >
                <label
                  className="bx--label"
                  htmlFor="percentage"
                >
                  Each Individual Investor Can
                  <br />
                  Own Up To of Outstanding Tokens
                </label>
                <div>
                  <TextInput
                    className="bx--text__input"
                    disabled={false}
                    helperText=""
                    id="percentage"
                    invalid={false}
                    invalidText=""
                    labelText=""
                    light={false}
                    onChange={[Function]}
                    onClick={[Function]}
                    placeholder="–"
                    type="text"
                  />
                  <Button
                    className="apply-percentage-btn"
                    disabled={true}
                    iconDescription="Provide icon description if icon is used"
                    kind="primary"
                    onClick={[Function]}
                    small={false}
                    tabIndex={0}
                    type="button"
                  >
                    Apply
                  </Button>
                </div>
              </div>
            </div>
            <div
              className="pui-clearfix"
            />
          </div>
        </div>
      </GridCol>
      <GridCol
        gridSpan={
          Array [
            12,
            12,
            6,
            6,
          ]
        }
      >
        <div
          id="compliance"
        >
          <br />
          <div
            className="pui-page-box compliance-form"
          >
            <h1
              className="pui-h1"
            >
              Whitelist Management
            </h1>
            <div
              className="whitelist-settings"
            >
              <div
                className="bx--form-item"
              >
                <label
                  className="bx--label"
                  htmlFor="whitelistToggle"
                >
                  Enable Third-Party Whitelist Management
                </label>
                <Toggle
                  defaultToggled={false}
                  id="whitelistToggle"
                  labelA="Off"
                  labelB="On"
                  onToggle={[Function]}
                />
              </div>
              <div
                className="bx--form-item"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <Connect(WhitelistTable) />
              </div>
            </div>
            <div
              className="pui-clearfix"
            />
          </div>
        </div>
      </GridCol>
    </GridRow>
    <GridRow
      gridGap={4}
    >
      <GridCol
        gridSpan={
          Array [
            12,
            12,
            12,
            12,
          ]
        }
      >
        <div
          id="compliance"
        >
          <br />
          <div
            className="pui-page-box compliance-form"
          >
            <h1
              className="pui-h1"
            >
              Manual Trade Approvals
            </h1>
            <p>
              Allows you to pre-approve token trades between two wallet addresses for a period of time. You can always edit and remove trade approvals. Note that these approvals supercede all other exemptions and restrictions for a particular wallet.
            </p>
            <div
              className="whitelist-settings"
            >
              <div
                className="bx--form-item"
              >
                <label
                  className="bx--label"
                  htmlFor="approvalToggle"
                >
                  Enable Manual Trade Approvals
                </label>
                <Toggle
                  defaultToggled={false}
                  id="approvalToggle"
                  labelA="Off"
                  labelB="On"
                  onToggle={[Function]}
                  toggled={true}
                />
              </div>
              <div
                className="bx--form-item"
                style={Object {}}
              >
                <Connect(ApprovalTable) />
              </div>
            </div>
            <div
              className="pui-clearfix"
            />
          </div>
        </div>
      </GridCol>
    </GridRow>
  </Grid>
</Page>
`;

exports[`Compliance Page should render NotFoundPage when there is no token 1`] = `<NotFoundPage />`;

exports[`Compliance Page should render correctly when token address is present 1`] = `
<Page
  py="xl"
  title="Compliance – Polymath"
>
  <Connect(Progress) />
  <Grid
    gridGap={4}
  >
    <GridRow
      gridGap="0px"
    >
      <GridCol
        gridSpan={9}
      >
        <h1
          className="pui-h1"
        >
          Token Compliance
        </h1>
        <h3
          className="pui-h3"
        >
          Manage compliance requirements of your tokens. You can import and export your whitelist, manage ownership percentages, add 3rd party whitelist managers, and manage approval of token transfers between wallets.
        </h3>
      </GridCol>
      <GridCol
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <Button
          className="import-whitelist-btn"
          disabled={false}
          icon="upload"
          iconDescription="Provide icon description if icon is used"
          kind="primary"
          onClick={[Function]}
          small={false}
          tabIndex={0}
          type="button"
        >
          Import Whitelist
        </Button>
      </GridCol>
      <GridCol
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <Connect(ImportWhitelistModal)
          isOpen={false}
          onClose={[Function]}
          onSubmit={[Function]}
        />
        <Button
          className="import-whitelist-btn"
          disabled={false}
          icon="download"
          iconDescription="Provide icon description if icon is used"
          kind="secondary"
          onClick={[Function]}
          small={false}
          tabIndex={0}
          type="button"
        >
          Export Whitelist
        </Button>
      </GridCol>
      <GridCol
        style={
          Object {
            "alignSelf": "center",
            "justifySelf": "left",
          }
        }
      >
        <OverflowMenu
          ariaLabel="list of options"
          direction="bottom"
          flipped={true}
          floatingMenu={true}
          iconDescription="open and close list of options"
          menuOffset={[Function]}
          menuOffsetFlip={[Function]}
          onClick={[Function]}
          onClose={[Function]}
          onKeyDown={[Function]}
          onOpen={[Function]}
          open={false}
          style={
            Object {
              "float": "right",
            }
          }
          tabIndex={0}
        >
          <OverflowMenuItem
            disabled={false}
            hasDivider={false}
            isDelete={false}
            itemText="Pause All Transfers"
            onClick={[Function]}
          />
        </OverflowMenu>
      </GridCol>
    </GridRow>
    <GridRow
      gridGap={4}
    >
      <GridCol
        gridSpan={
          Array [
            12,
            12,
            6,
            6,
          ]
        }
      >
        <div
          id="compliance"
        >
          <br />
          <div
            className="pui-page-box compliance-form"
          >
            <h1
              className="pui-h1"
            >
              Ownership Restrictions
            </h1>
            <div
              className="compliance-settings"
            >
              <div
                className="bx--form-item"
              >
                <label
                  className="bx--label"
                  htmlFor="percentageToggle"
                >
                  Enable Ownership Restrictions
                </label>
                <Toggle
                  defaultToggled={false}
                  id="percentageToggle"
                  labelA="Off"
                  labelB="On"
                  onToggle={[Function]}
                  toggled={false}
                />
              </div>
              <div
                className="bx--form-item"
                style={Object {}}
              >
                <label
                  className="bx--label"
                  htmlFor="percentage"
                >
                  Each Individual Investor Can
                  <br />
                  Own Up To of Outstanding Tokens
                </label>
                <div>
                  <TextInput
                    className="bx--text__input"
                    disabled={false}
                    helperText=""
                    id="percentage"
                    invalid={false}
                    invalidText=""
                    labelText=""
                    light={false}
                    onChange={[Function]}
                    onClick={[Function]}
                    placeholder="–"
                    type="text"
                  />
                  <Button
                    className="apply-percentage-btn"
                    disabled={true}
                    iconDescription="Provide icon description if icon is used"
                    kind="primary"
                    onClick={[Function]}
                    small={false}
                    tabIndex={0}
                    type="button"
                  >
                    Apply
                  </Button>
                </div>
              </div>
            </div>
            <div
              className="pui-clearfix"
            />
          </div>
        </div>
      </GridCol>
      <GridCol
        gridSpan={
          Array [
            12,
            12,
            6,
            6,
          ]
        }
      >
        <div
          id="compliance"
        >
          <br />
          <div
            className="pui-page-box compliance-form"
          >
            <h1
              className="pui-h1"
            >
              Whitelist Management
            </h1>
            <div
              className="whitelist-settings"
            >
              <div
                className="bx--form-item"
              >
                <label
                  className="bx--label"
                  htmlFor="whitelistToggle"
                >
                  Enable Third-Party Whitelist Management
                </label>
                <Toggle
                  defaultToggled={false}
                  id="whitelistToggle"
                  labelA="Off"
                  labelB="On"
                  onToggle={[Function]}
                />
              </div>
              <div
                className="bx--form-item"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <Connect(WhitelistTable) />
              </div>
            </div>
            <div
              className="pui-clearfix"
            />
          </div>
        </div>
      </GridCol>
    </GridRow>
    <GridRow
      gridGap={4}
    >
      <GridCol
        gridSpan={
          Array [
            12,
            12,
            12,
            12,
          ]
        }
      >
        <div
          id="compliance"
        >
          <br />
          <div
            className="pui-page-box compliance-form"
          >
            <h1
              className="pui-h1"
            >
              Manual Trade Approvals
            </h1>
            <p>
              Allows you to pre-approve token trades between two wallet addresses for a period of time. You can always edit and remove trade approvals. Note that these approvals supercede all other exemptions and restrictions for a particular wallet.
            </p>
            <div
              className="whitelist-settings"
            >
              <div
                className="bx--form-item"
              >
                <label
                  className="bx--label"
                  htmlFor="approvalToggle"
                >
                  Enable Manual Trade Approvals
                </label>
                <Toggle
                  defaultToggled={false}
                  id="approvalToggle"
                  labelA="Off"
                  labelB="On"
                  onToggle={[Function]}
                />
              </div>
              <div
                className="bx--form-item"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <Connect(ApprovalTable) />
              </div>
            </div>
            <div
              className="pui-clearfix"
            />
          </div>
        </div>
      </GridCol>
    </GridRow>
  </Grid>
</Page>
`;
